### PR TITLE
OSDP Kconfig enhancements

### LINKED ISSUE #28114: subsys: OSDP forces SERIAL=y
**Describe the bug**
OSDP subsys forces SERIAL Kconfig symbol in its definition:
```
menuconfig OSDP
	bool "Open Supervised Device Protocol (OSDP) driver"
	select RING_BUFFER
	select SERIAL
	select SERIAL_SUPPORT_INTERRUPT
	help
	  Add support for Open Supervised Device Protocol (OSDP)
```

One of the consequence is that sanitycheck filtering deos not operate and sample run on platform not supporting SERIAL:
```
tests:
  sample.mgmt.osdp:
    tags: osdp
    filter: dt_compat_enabled_with_alias("gpio-leds", "led0") and CONFIG_SERIAL
    harness: osdp
```

When running sanitychcek, test is ran on stm32h747i_disco_m4, which does not support SERIAL by default
:
```
$ sanitycheck -b -p stm32h747i_disco_m4 -T samples/subsys/mgmt/osdp
Deleting output directory /local/mcu/zephyrproject/zephyr/sanity-out
INFO    - JOBS: 16
INFO    - Building initial testcase list...
INFO    - 1 test configurations selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...

ERROR   - stm32h747i_disco_m4       samples/subsys/mgmt/osdp/sample.mgmt.osdp          FAILED: Build failure
ERROR   - see: /local/mcu/zephyrproject/zephyr/sanity-out/stm32h747i_disco_m4/samples/subsys/mgmt/osdp/sample.mgmt.osdp/build.log
INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    1
INFO    - 0 of 1 tests passed (0.00%), 1 failed, 0 skipped with 0 warnings in 6.44 seconds
INFO    - In total 1 test cases were executed on 1 out of total 284 platforms (0.35%)
INFO    - 0 tests executed on platforms, 1 tests were only built.
```

**To Reproduce**
```
$ sanitycheck -b -p stm32h747i_disco_m4 -T samples/subsys/mgmt/osdp
```

**Expected behavior**
OSDP should follow the approach used for similar subsy, such as modem:
```
config MODEM_RECEIVER
	bool "Enable modem receiver helper driver"
	depends on SERIAL_SUPPORT_INTERRUPT
	select UART_INTERRUPT_DRIVEN
	select RING_BUFFER
	help
```

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Logs and console output**
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**

### PR DESCRIPTION
**Pull request by sidcha**
- Make CP/PD mode selection as first entry in Kconfig
- Change `select SERIAL` to `imply SERIAL_SUPPORT_INTERRUPT`
- Add MAINTAINERS.yml entry for OSDP

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/28114

CC: @erwango

### CODE REVIEW

File: subsys/mgmt/osdp/Kconfig [hunk 1]
```diff
@@ -14,6 +14,19 @@ menuconfig OSDP
 
 if OSDP
 
+choice
+	prompt "OSDP Mode of Operation"
+	default OSDP_MODE_PD
```
Comment (erwango) [inline subsys/mgmt/osdp/Kconfig:19]: Default value will be systematically the choice proposition.
So we usually put propositions in such order that the first one is the default and don't use explicit 'default'.
Please reorder accordingly.

**Review by erwango — CHANGES_REQUESTED** [review]

File: subsys/mgmt/osdp/Kconfig (hunk 1 above)
Comment (cbsiddharth) [reply subsys/mgmt/osdp/Kconfig]: Done. Thanks.

**Review by erwango — APPROVED** [review]

**Comment by cbsiddharth** [comment]
@erwango, switched to using `imply` as suggested by @galak and now osdp gets filtered out of platform that don't support CONFIG_SERIAL. Please have a look now.

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -846,6 +846,18 @@ MCU Manager:
     labels:
         - "area: mcumgr"
 
+OSDP:
+    # Provisional entry, subject to approval
+    status: maintained
+    maintainers:
```
Comment (MaureenHelm) [inline MAINTAINERS.yml:852]:
```suggestion
    collaborators:
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 2 above)
Comment (cbsiddharth) [reply MAINTAINERS.yml:852]: Is someone else planning on taking up the maintership of this code?

Comment (erwango) [reply MAINTAINERS.yml:852]: @cbsiddharth maintainers have to be approved by TSC, and this is not the case yet here.

Comment (cbsiddharth) [reply MAINTAINERS.yml]: Okay, changed to `collaborators`.

Comment (MaureenHelm) [reply MAINTAINERS.yml]: Some more context [here](https://docs.zephyrproject.org/latest/contribute/project_roles.html)

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**